deployment/main.tf

```diff
@@ -0,0 +1,135 @@
+variable "gcp_project" {
+  type = string
+}
+
+variable "region" {
+  type    = string
+  default = "us-west1"
+}
+
+provider "google" {
+  project = var.gcp_project
+}
+
+provider "google-beta" {
+  project = var.gcp_project
+}
+
+# Enable required API in the project
+
+resource "google_project_service" "container-api" {
+  service = "container.googleapis.com"
+}
+
+resource "google_project_service" "spanner-api" {
+  service = "spanner.googleapis.com"
+}
+
+# Main cluster definition
+
+## Force recent version for Kubernetes master
+data "google_container_engine_versions" "gke-ver" {
+  location       = var.region
+  version_prefix = "1.14."
+}
+
+resource "google_container_cluster" "trillian-cluster" {
+  provider           = "google-beta"
+  name               = "cluster"
+  location           = var.region
+  node_version       = "${data.google_container_engine_versions.gke-ver.latest_node_version}"
+  min_master_version = "${data.google_container_engine_versions.gke-ver.latest_node_version}"
+
+  initial_node_count = 3
+
+  node_config {
+    machine_type = "n1-standard-2"
+    image_type   = "COS"
+
+    workload_metadata_config {
+      node_metadata = "GKE_METADATA_SERVER"
+    }
+    oauth_scopes = [
+      "https://www.googleapis.com/auth/logging.write",
+      "https://www.googleapis.com/auth/monitoring",
+      "https://www.googleapis.com/auth/devstorage.read_only"
+    ]
+  }
+
+  workload_identity_config {
+    identity_namespace = "${var.gcp_project}.svc.id.goog"
+  }
+}
+
+# Spanner instance
+
+locals {
+  trillian_ddl = file("${path.module}/../storage/cloudspanner/spanner.sdl")
+}
+
+resource "google_spanner_instance" "trillian-spanner" {
+  name         = "trillian-spanner"
+  display_name = "Trillian Spanner Instance"
+  config       = "regional-${var.region}"
+  num_nodes    = 1
+}
```

Comment:
I think you need to add a dependency on `google_project_service.spanner-api` here:
```terraform
depends_on = [
  google_project_service.spanner-api,
]
```
Without it, this resource won't wait for the Spanner API to be enabled, resulting in errors the first time to run `terraform apply`. I think there are a few other resources that require APIs to be enabled here that don't express that dependency.